packages/effect/src/Stream.ts

```diff
@@ -4329,6 +4329,44 @@ export const zipLatest: {
   <A, E, R, A2, E2, R2>(self: Stream<A, E, R>, that: Stream<A2, E2, R2>): Stream<[A, A2], E | E2, R | R2>
 } = internal.zipLatest
 
+/**
+ * Zips multiple streams so that when a value is emitted by any of the streams,
+ * it is combined with the latest values from the other streams to produce a result.
+ *
+ * Note: tracking the latest value is done on a per-chunk basis. That means
+ * that emitted elements that are not the last value in chunks will never be
+ * used for zipping.
+ *
+ * @example
+ * import { Stream } from "effect"
+ *
+ * const stream = Stream.zipLatestAll(
+ *     Stream.fromSchedule(Schedule.spaced('500 millis')),
+ *     Stream.fromSchedule(Schedule.spaced('1000 millis')),
+ *     Stream.fromSchedule(Schedule.spaced('2000 millis')),
+ * ).pipe(Stream.tap(Console.log))
+ *
+ * Effect.runPromise(Stream.runDrain(stream))
+ * // Output:
+ * [ 0, 0, 0 ]
+ * [ 1, 0, 0 ]
+ * [ 1, 1, 0 ]
+ * [ 2, 1, 0 ]
+ * [ 3, 1, 0 ]
+ * [ 3, 1, 1 ]
+ * // .....
+ *
+ * @since 2.0.0
+ * @category zipping
+ */
+export const zipLatestAll: <T extends Array<Stream<unknown>>>(
```

Comment:
```suggestion
export const zipLatestAll: <T extends ReadonlyArray<Stream<any, any, any>>>(
```

Otherwise you will run into assignability issues.